File: 11-mersenne-num/mersenne_gen.py
# Python program to generate mersenne numbers
# Created by M.V.Harish Kumar on 20-11-2023

n = int(input("Number of mersenne numbers to generate: "))
mersenne = []
for i in range(1, n+1):
    mersenne.append(2 ** i - 1)
print(f"The first {n} mersenne numbers are:")
print("\t", str(mersenne)[1:-1])

File: 11-mersenne-num/mersenne_gen.go
// Go program to generate mersenne numbers
// Written by M.V.Harish Kumar on 18-06-2025
package main

import (
	"fmt"
	"math"
	"strings"
)

func main() {
	var num uint
	var mersenneList []string

	fmt.Print("Number of mersenne numbers to generate: ")
	fmt.Scanln(&num)

	for i := range num {
		val := fmt.Sprint(math.Exp2(float64(i + 1)) - 1)
		mersenneList = append(mersenneList, val)
	}

	fmt.Printf("The first %d mersenne numbers are:\n", num)
	fmt.Println("\t", strings.Join(mersenneList, ", "))

}
